### PR TITLE
Moving platforms

### LINKED ISSUE #37: mario can insta-die from being squished?
if I remember correctly, Mario can instantly die even past invulnerability if he's somehow underneath a moving platform as it push him into the ground...this is probably true of every entity actually and also can happen in any direction a platform moves in. Might also not be a valid concern if no level has a moving platform which ever goes near a brick.

not sure how to best catch this 3+entity interaction or entirely sure that my memory is correct but it seems like a reasonable solution for a game that probably can't figure out the forces or whatever to figure out if mario should die or just get pushed sideways.

regardless, I'm like 99% sure that our current collision detection won't handle this probably rare situation correctly (mario will probably phase through the platform or the bricks).

edit: probably related to the issue in https://github.com/algorithm0r/SuperMarioBros/issues/38#issue-818601022
as it is also due to handling 3+ entity collision handling and this other bug makes me quite confident that this will be a bug even if its not effectively one due to not being relevant.

### PR DESCRIPTION
**Pull request by ghost**
# Pull Request Description

Please fill out the form below. For the checkbox, use [x] to have a checkmark. For example:

- [x] With a checkmark.
- [ ] Without a checkmark.

This should not affect the way whether your pull request is accepted or not. Think of it like a survey and documentation, so the reviewers and future developers can follow your PR easier.

1. Have you included *.gitignore* file? Or manually remove config and environment files (E.g: *node_modules* folder, Eclipse IDE files) from this submission? The answer should be **Yes** in this case.

   - [x] Yes
   - [ ] No

2. Did you fully complete the assigned task(s)?

   - [x] Fully completed
   - [ ] Partially completed

3. Did you introduce a new non-required feature in this PR? This feature is not an assigned task, but it is the one that you feel will make the game better. For example, reproduce a bug used by speed runner in an original version.

   - [ ] Yes
   - [x] No

4. Are you aware of any bug caused by your code that have not been fixed in this PR?

   - [x] Yes
   - [ ] No

5. Are you aware of any **potential bug**? You don't have to fix this, it is just a way to document. You may not face the bug during the development process, but theoretically it can happens in edge cases. For example, memory leak or a player falls out of map.

   - [ ] Yes
   - [x] No

## Related issue(s)

- #13

## Added

- Implemented moving platforms using Victor's code as a base.

## Bugs

1. Not-fixed bugs (only answer when you chose **Yes** in question 4):
   - In some cases, it is possible to reach the top of a platform if Mario collides with the side of it.

## Contributors and code author(s)

- Kyle Oslin

## Final check

Have you fully filled out this document, and add a label **PR: NEED REVIEW** to this pull request? Answer *Yes* (as it should be) means you are good to go! Thanks for the contribution.

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by hunghvu** [comment]
Because this is built upon #29. **DO NOT MERGE until after #29 is merged.**

**Review by hunghvu — DISMISSED** [review]
The platforms does not collide with Mario when he hits them from the side or bottom (as shown in the video, pay attention to the mini map too). This result in Mario going through the platform, I consider this a potential game breaking bug so change is requested. 

Besides, it appears that there must be 2 platforms on screen at a time. You can see that [here](https://www.youtube.com/watch?v=fJPdDCsROik&ab_channel=JannisXxL). 
image 


https://user-images.githubusercontent.com/62678592/109266256-a58d7400-77bc-11eb-936c-805d46ee09ae.mp4

File: mario.js
Comment (BenjaminDeJager) [inline mario.js:301]: fyi, irrc mario can instantly die if hes underneith a platform as it pushs himinto the ground...or maybe thats just the newer games? seems likely to be in the older games as it seems like the easiest solution.

as this has to do with more "advanced" bounding-box/collision detection (involves 3+ entity's), it should probably be a seperate issue

**Comment by BenjaminDeJager** [comment]
I see hung has requested changes so I will hold off on my review till this has seen another edit.

**Review by hunghvu — APPROVED** [review]

**Review by algorithm0r — APPROVED** [review]

**Pull request merged**
